### PR TITLE
teapot: init at 2.3.0

### LINKED ISSUE #147213: Package Request:  Teapot, a compact, type-safe, 3D spreadsheet
**Project description**

Teapot is an interesting project that implements a compact spreadsheet in roughly 140k, with the following features:

- **UI:**  Ncurses terminal UI, FLTK windowed GUI, and a batch processing mode all available.
- **Cell Syntax:**  Cell locations represented as lists of coordinates.  Eg, instead of "B1" or "R1C2", cells are referenced as "@(1,2,0)".
- **Three Dimensions:**  This 3-coordinate style adds a third dimension to the spreadsheet, reducing the need for hiding columns or rows of intermediate results as is sometimes the case in traditional spreadsheets, among other things.
- **Relative References:**  This also enables both relative cell and relative formula references that make refactoring the spreadsheet easier.
- **Type-Safety:**  All values have a type, and all operations and formulas check types.  No more adding a float to a string and getting a (probably undesirable) result.
- **Iterative Expressions:**  Modeled on clocked expressions, these solve problems with circular dependencies in traditional spreadsheets and effectively make teapot Turing-complete.
- **Platform-Independent File format:**  Defined by XDR and portable across machines, especially floating point representation that may differ across platforms
- **Legacy File format support:**  Portable ASCII file format, standard CSV, SpreadsheetCalculator SC and Lotus 1-2-3 WK1 format.
- **Extensibility:**  Easily extended by new functions in C.
- **Internationalization:**  Can be internationalized and comes with English, Dutch, German and Polish message catalogs.
- **Interoperability:**  Interfaces to troff/tbl, LATEX, HTML and CSV which are well-suited for further processing, be it graphing, typesetting, accounting or anything else.

**Metadata**

* homepage URL:  https://www.syntax-k.de/projekte/teapot/
* source URL:  https://www.syntax-k.de/projekte/teapot/ (bottom of page)
* license: GPL3
* platforms: unix, linux, darwin

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Motivation for this change

Closes #147213 (but not so much)

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 147440` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>teapot</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

**Review by thiagokokada — COMMENTED** [review]
> I was feeling like as a first time vim user, without a clue how to exit the program.
> 10/10 would use it again.

File: pkgs/applications/office/teapot/default.nix [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      Another interesting feature is its modern approach to spread sheet theory:
+      It sports true three-dimensional tables and iterative expressions. And
+      since it breaks compatibility with the usual notions of big spreadsheet
+      packages, it can also throw old syntactic cruft over board which many
+      spreadsheets still inherit from the days of VisiCalc on ancient CP/M
+      systems.
+    '';
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ AndersonTorres ];
+    platforms = platforms.unix;
```
[the remark this answers is not in the document]
Comment (AndersonTorres) [reply pkgs/applications/office/teapot/default.nix:69]: I don't like to use default values here in standalone programs.
Then only sensible places to "go with default" are packagesets like `rubyPackages`, `pythonPackages` etc.

**Comment by AndersonTorres** [comment]
Well, better than this only if I send patches to the original project. 

Maybe another day...

**Comment by AndersonTorres** [comment]
> > I was feeling like as a first time vim user, without a clue how to exit the program.
> > 10/10 would use it again.

poweroff button to the rescue!

**Pull request merged**